### ISSUE TITLE
Task: Set up local environment

### ISSUE THREAD

**Issue by mtreacy002** [anitab-org/bridge-in-tech-web#218]
## Description
As a contributor
I need to set up local environment for bridge-in-tech-web and send screenshots,
so that I can provide proof of the same.

## Mocks
NA

### Update [Required]
- [ ] Set up local environment
- [ ] Provide a **gif** of a successful login attempt that must show the change on main navbar before and after login.
 
      **Before login**
<img width="1210" alt="Screen Shot 2021-03-30 at 8 46 13 am" src="https://user-images.githubusercontent.com/29667122/112904384-9c355580-9134-11eb-8147-ba811c2a3d02.png">

      **After login**
<img width="1216" alt="Screen Shot 2021-03-30 at 8 46 35 am" src="https://user-images.githubusercontent.com/29667122/112904442-b1aa7f80-9134-11eb-9be9-59b6b79bea52.png">

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
2-3 hours

**Additional note:**
**1. This is a compulsory task that must be completed before contributors can claim other Open Source Hack issues.
2. The PR that is sibmitted for this task will counted towards the `Documentation/Training` category of Open Source Hack.**. To do this you can create a new branch called `setup` and add an end of line comment inside the index.js (e.g. //Compulsory task - Setup environment) so git pick up a change in the code base. Then once you submit the PR, inside that PR description you can add the gif under the "How this has been tested").

**Comment by Rahulm2310** [comment]
@mtreacy002 I don't think this should be a proof of setup as anyone can mess up with HTML and CSS from the developer console and take a screenshot. We need a GIF where user inputs his credentials and then login.

Also, please add this note in the issue description. 
#### NOTE: This issue is available for everyone. Please share a GIF  below in comments showing the complete login process.

**Comment by mtreacy002** [comment]
Done, @Rahulm2310 , I've changed the issue description to state these conditions (the gif and must be done before other OSH issue).

**Comment by naveen8801** [comment]
Hi @mtreacy002 @Rahulm2310  , as a beginner it would be helpful for me to work on this , can you assign it to me , :-)

**Comment by decon-harsh** [comment]
@naveen8801 

This issue is available for everyone.This is a compulsory task and must be completed before contributor can claim other Open Source Hack issue.
Moreover You have been already assigned #214.

**Comment by naveen8801** [comment]
> @naveen8801 
> 
> This issue is available for everyone.This is a compulsory task and must be completed before contributor can claim other Open Source Hack issue
> 

I got it @decon-harsh  , thanks :-)

**Comment by shades-7** [comment]
hey I tried setting up this environment, I am running BIT API at localhost 5000, and mentorship system API at 4000 while registering i am getting error 500

**Comment by decon-harsh** [comment]
> hey I tried setting up this environment, I am running BIT API at localhost 5000, and mentorship system API at 4000 while registering i am getting error 500

Hey , it would be better if you join our zulip and ask your doubts regarding the issue there. https://anitab-org.zulipchat.com/login/

**Comment by shades-7** [comment]
> > hey I tried setting up this environment, I am running BIT API at localhost 5000, and mentorship system API at 4000 while registering i am getting error 500
> 
> Hey , it would be better if you join our zulip and ask your doubts regarding the issue there. https://anitab-org.zulipchat.com/login/

done thank you

**Comment by shades-7** [comment]
I have completed this task, below is a gif. Many thanks to @decon-harsh 

![BIT](https://user-images.githubusercontent.com/58786336/113373249-101b6c00-9388-11eb-8aa1-362049a095de.gif)

**Comment by decon-harsh** [comment]
If anyone else is facing any issue, join zulip and feel free to ask it there.

**Comment by mtreacy002** [comment]
> I have completed this task, below is a gif. Many thanks to @decon-harsh
> 
>  BIT.mp4

@shades-7, can you upload a gif instead? it seems like the .mp4 is not recognised format here.

**Comment by shades-7** [comment]
> > I have completed this task, below is a gif. Many thanks to @decon-harsh
> > BIT.mp4
> 
> @shades-7, can you upload a gif instead? it seems like the .mp4 is not recognised format here.

done

**Comment by Rahulm2310** [comment]
@shades-7 approved :100:  . Feel free to claim other issues now 😉

**Comment by sdew021** [comment]
@Rahulm2310 Am I supposed be assigned to this issue to complete this task? If yes, please assign it to me

**Comment by Rahulm2310** [comment]
@sdew021 You need not to be assigned to this issue. This issue is available for everyone. Complete it and post a GIF below. 😊

**Comment by sdew021** [comment]
@Rahulm2310 okay, there's another issue, when I try to signup, I get this error message "The server is currently unavailable. Try again later"

**Comment by Rahulm2310** [comment]
@sdew021 you need to setup bridge-in-tech-backend and ms-server (from @mtreacy002 's fork repo) locally and run them as we don't have a remote backend deployed. You can go through the instructions on readme and wiki. Feel free to post your queries on our Zulip channel. 😊

**Comment by sdew021** [comment]
okay, thankyou😊

**Comment by sdew021** [comment]
So I need to follow all the instructions from https://github.com/anitab-org/bridge-in-tech-backend/blob/develop/.github/ENV_SETUP_INSTRUCTION.md this file only right ?

**Comment by decon-harsh** [comment]
> @sdew021 you need to setup bridge-in-tech-backend and mentorship-backend locally and run them as we don't have a remote backend deployed. You can go through the instructions on readme and wiki. Feel free to post your queries on our Zulip channel. 😊

Hey @Rahulm2310 Can you edit this message. It would be better if you specify Ms-server of Maya Treacy's Fork instead of mentorship backend as it might confuse some new comers .

**Comment by mtreacy002** [comment]
@shades-7 , don't forget to submit a PR for your local setup so it is counted towards the `Documentation/Training` category of your Open Source Hack contribution. That is one out of 4 categories down to be eligible for the OSH certificate 😉

**Comment by shades-7** [comment]
> @shades-7 , don't forget to submit a PR for your local setup so it is counted towards the `Documentation/Training` category of your Open Source Hack contribution. That is one out of 4 categories down to be eligible for the OSH certificate

But there won't be any file change how to send PR for this.

**Comment by naveen8801** [comment]
Hii @Rahulm2310 , @mtreacy002  

I have successfully completed this task , thanks a lot to @jalajcodes 👍 , attaching gif below :

![success](https://user-images.githubusercontent.com/54985099/113488089-8b406780-94d9-11eb-814d-182e5b189bbd.gif)

Should i make a PR for this one also ?

**Comment by Rahulm2310** [comment]
@naveen8801 Yes! Please submit a PR with branch called `setup` and add this GIF in the PR description. This pr will be counted towards the `Documentation/Training` category which means you will achieve one out of 4 categories you must complete for OSH.

cc: @mtreacy002

**Comment by naveen8801** [comment]
> @naveen8801 Yes! Please submit a PR with branch called `setup` and add this GIF in the PR description. This pr will be counted towards the `Documentation/Training` category which means you will achieve one out of 4 categories you must complete for OSH.
> 
> cc: @mtreacy002

@Rahulm2310  , but how to make a PR , because there is no file change .

**Comment by mtreacy002** [comment]
@naveen8801 , I added the tips on how to submit a pr on this issue description. Let me know if you still have doubt.

**Comment by naveen8801** [comment]
> @naveen8801 , I added the tips on how to submit a pr on this issue description. Let me know if you still have doubt.

Thanks @mtreacy002  , i have made a PR , please review :-)

**Comment by mtreacy002** [comment]
Closing this now since Open Source Hack has ended.
